hydrolib/core/dflowfm/extold/models.py

```diff
@@ -0,0 +1,571 @@
+
+        if lower_value in list(ExtOldQuantity):
+            return ExtOldQuantity(lower_value)
+
+        supported_value_str = ", ".join(([x.value for x in ExtOldQuantity]))
+        raise ValueError(
+            f"QUANTITY '{value_str}' not supported. Supported values: {supported_value_str}"
+        )
+
+    @validator("operand", pre=True)
+    def validate_operand(cls, value):
+        if isinstance(value, Operand):
+            return value
+
+        if isinstance(value, str):
+
+            for operand in Operand:
+                if value.lower() == operand.value.lower():
+                    return operand
+
+            supported_value_str = ", ".join(([x.value for x in Operand]))
+            raise ValueError(
+                f"OPERAND '{value}' not supported. Supported values: {supported_value_str}"
+            )
+
+        return value
+
+    @root_validator(skip_on_failure=True)
+    def validate_forcing(cls, values):
+        class _Field:
+            def __init__(self, key: str) -> None:
+                self.alias = cls.__fields__[key].alias
+                self.value = values[key]
+
+        def raise_error_only_allowed_when(
+            field: _Field, dependency: _Field, valid_dependency_value: str
+        ):
+            error = f"{field.alias} only allowed when {dependency.alias} is {valid_dependency_value}"
+            raise ValueError(error)
+
+        def only_allowed_when(
+            field: _Field, dependency: _Field, valid_dependency_value: Any
+        ):
+            """This function checks if a particular field is allowed to have a value only when a dependency field has a specific value."""
+
+            if field.value is None or dependency.value == valid_dependency_value:
+                return
+
+            raise_error_only_allowed_when(field, dependency, valid_dependency_value)
+
+        quantity = _Field("quantity")
+        varname = _Field("varname")
+        sourcemask = _Field("sourcemask")
+        filetype = _Field("filetype")
+        method = _Field("method")
+        extrapolation_method = _Field("extrapolation_method")
+        maxsearchradius = _Field("maxsearchradius")
+        value = _Field("value")
+        factor = _Field("factor")
+        ifrctype = _Field("ifrctyp")
+        averagingtype = _Field("averagingtype")
+        relativesearchcellsize = _Field("relativesearchcellsize")
+        extrapoltol = _Field("extrapoltol")
+        percentileminmax = _Field("percentileminmax")
+        area = _Field("area")
+        nummin = _Field("nummin")
+
+        only_allowed_when(varname, filetype, ExtOldFileType.NetCDFGridData)
+
+        if sourcemask.value.filepath is not None and filetype.value not in [
+            ExtOldFileType.ArcInfo,
+            ExtOldFileType.CurvilinearData,
+        ]:
+            raise_error_only_allowed_when(
+                sourcemask, filetype, valid_dependency_value="4 or 6"
+            )
+
+        if (
```

Comment:
Let's have a very short discussion if this works as intented. It always confuses me 😊

Reply:
Nevermind, this should work! 😊